packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
+  /// Function that is called when the card stack runs out of cards.
   final VoidCallback? onEnd;
 
-  /// function that gets triggered when the swiper is disabled
+  /// Function that is called when a user attempts to swipe but swiping is
+  /// disabled.
   final VoidCallback? onTapDisabled;
 
-  /// function that gets called with the boolean true when the last card gets unswiped and with the boolean false when there is no card to unswipe
+  /// Function that gets called when an unswipe is triggered.
+  ///
+  /// This is called with a boolean flag that is true if the unswipe was
+  /// successful and false if it was not (there was no swipe to undo).
   final OnUnSwipe? unswipe;
 
-  /// direction in which the card gets swiped when triggered by controller, default set to right
-  final AppinioSwiperDirection direction;
+  /// The default direction in which the card gets swiped when triggered by
+  /// controller.
+  ///
+  /// Defaults to [AxisDirection.right].
+  final AxisDirection defaultDirection;
 
   const AppinioSwiper({
     Key? key,
-    required this.cardsBuilder,
-    required this.cardsCount,
+    required this.cardBuilder,
+    required this.cardCount,
     this.controller,
-    this.padding = const EdgeInsets.symmetric(horizontal: 20, vertical: 25),
     this.duration = const Duration(milliseconds: 200),
-    this.maxAngle = 30,
+    this.maxAngle = 15,
+    this.invertAngleOnBottomDrag = true,
     this.cardsSpacing = 40,
     this.threshold = 50,
-    this.backgroundCardsCount = 1,
+    this.backgroundCardCount = 1,
+    this.backgroundCardScale = .9,
+    this.backgroundCardOffset,
     this.isDisabled = false,
     this.loop = false,
-    this.swipeOptions = const AppinioSwipeOptions.all(),
-    this.allowUnswipe = true,
-    this.unlimitedUnswipe = false,
+    this.swipeOptions = const SwipeOptions.all(),
     this.onTapDisabled,
     this.onSwipe,
     this.onSwipeCancelled,
-    this.onSwiping,
+    this.onCardPositionChanged,
     this.onEnd,
     this.unswipe,
-    this.direction = AppinioSwiperDirection.right,
-  })  : assert(maxAngle >= 0 && maxAngle <= 360),
+    this.defaultDirection = AxisDirection.right,
+  })  : assert(maxAngle >= 0),
         assert(threshold >= 1 && threshold <= 100),
-        assert(direction != AppinioSwiperDirection.none),
         super(key: key);
 
   @override
   State createState() => _AppinioSwiperState();
 }
 
 class _AppinioSwiperState extends State<AppinioSwiper>
-    with SingleTickerProviderStateMixin {
-  double _left = 0;
-  double _top = 0;
-  double _total = 0;
-  double _angle = 0;
-  double _maxAngle = 0;
-  double _scale = 0.9;
-  double _difference = 40;
-  double _backgroundCardsDifference = 40;
-  final double _backgroundCardsScaleDifference = 0.1;
-  int currentIndex = 0;
-
-  int _swipeType = 0; // 1 = swipe, 2 = unswipe, 3 = goBack
-  bool _tapOnTop = false; //position of starting drag point on card
-
-  late AnimationController _animationController;
```

Comment:
I eliminated all cases where a variable wasn't initialized on declaration. This massively helps maintainability/readability.